custom_components/tuya_local/devices/geyserwise_mws_water_heater.yaml

```diff
@@ -0,0 +1,217 @@
+name: Geyserwise MWS
+products:
+  - id: qwbyia2yr5ip9y4v
+    manufacturer: Geyserwise
+    model: MWS
+    name: Solar MPPT Geyser
+entities:
+  - entity: water_heater
+    dps:
+      - id: 1
+        type: boolean
+        name: operation_mode
+        mapping:
+          - dps_val: false
+            value: "Off"
+          - dps_val: true
+            value: "On"
+      - id: 2
+        type: string
+        name: away_mode
+        mapping:
+          - dps_val: "Timer"
+            value: false
+          - dps_val: "Holiday"
+            value: true
+      - id: 10
+        type: integer
+        name: current_temperature
+        readonly: true
```

Comment:
Readonly is only needed in exceptional cases where the HA entity model expects a value to be read-write but the device does not support that and there is a strong reason not to change the entity mapping. Here, current_temperature is always expected to be read-only.